File: jukengkengparent/jukengkeng-sellergoods-interface/src/main/java/com/jukengkeng/sellergoods/service/SpecificationService.java
package com.jukengkeng.sellergoods.service;

import com.jukengkeng.pojo.TbSpecification;
import com.jukengkeng.pojogroup.Specification;
import entity.pageResult;

import java.util.List;
import java.util.Map;

public interface SpecificationService {
    //规格分页方法
    pageResult findPageList(Integer pageNum, Integer pageSize, TbSpecification tbSpecification);

    //新增规格及规格选项
    void addSpecification(Specification specification);

    //删除规格及规格选项
    void delSpecifciation(Long[] id);

    //根据ID查询规格及规格选项
    Specification findSpecificationByID(Long id);

    //更新规格及规格选项
    void updateSpecification(Specification specification);

    //获取规格下拉列表
    List<Map> selectSpecificationOption();
}

File: jukengkengparent/jukengkeng-manager-web/src/main/webapp/js/myAction/service/specificationService.js
app.service("specificationService", function ($http) {
    //分页查询service
    this.queryList = function (pageNum, pageSize, searchEntity) {
        return $http.post("../specification/findPageList.do?pageNum=" + pageNum + "&pageSize=" + pageSize, searchEntity);
    }

    //新增规格及规格选项
    this.addSpecification = function (specification) {
        return $http.post("../specification/add.do", specification);
    }

    //删除规格及规格选项
    this.delSpecifciation = function (selectIds) {
        return $http.post("../specification/del.do?id=" + selectIds);
    }

    //根据ID查询规格及规格选项
    this.findSpecificaitonByID = function (id) {
        return $http.post("../specification/findSpecificationByID.do?id=" + id);
    }

    //更新规格及规格选项
    this.updateSpecification = function (specification) {
        return $http.post("../specification/updateSpecification.do", specification);
    }
    
    //获取规格下拉列表数据
    this.selectSpecificationOption = function () {
        return $http.post("../specification/selectSpecificationOption.do");
    }
});